0162-semantic-convention-resource-stability.md

```diff
@@ -0,0 +1,110 @@
+# Resource Semantic Convention Stabiltiy Requirements
+
+Outline what changes are "safe" to perform for resource semantic conventions
+and which require major version bumps.
+
+## Motivation
+
+This outlines acceptable changes to semantic conventions for resources, allowing
+growth and evolution over time.
+
+While Telemetry Schemas propose a mechanism to define changes to semantic
+conventions in a way that users can evolve telemetry, this document first
+attempts to define what is considered a compatible change.
+
+## Explanation
+
+We take the assumption that `Resource`, as it exists today, is the identity of
+an SDK and is leveraged as such in a lot of assumptions throughout open
+telemetry. Specifically, if an SDK that used to produce a `Resource` of
+attributes XYZ instead produces ABC, these metrics would be considered from a
+different source. To that end, if a Metric produced on version x.y of
+OpenTelemetry SDK is incompatible with one produced on version x.{y+1} because
+of `Resource` changes, this would be considered breaking.
+
+This means the following:
+
+- `Resource` detectors should provide a comprehensive set of attributes in
+  all or nothing fashion. if a resource detector produced attributes XYZ for
+  a given compute, it should continue to do so for compatible versions.
+- Resource attributes (e.g. `cloud.provider`) may have new possible values
+  defined as long as there is zero overlap with how the previous values were
+  computed.  For example, `acme-cloud` would never be inferred where previously
+  `aws` was inferred.
+  *Note: if `aws` was inferred incorrectly on "Acme cloud" previously, that would not be a violation.*
+- Adding new Resource attribute semantic conventions is allowed, however
+  existing resource detectors MUST NOT use them, only new detectors would be
+  allowed.
+
+If we assume capabilities of
+[Telemetry Schema conversion](https://github.com/open-telemetry/oteps/pull/161), 
+then we can relax some restrictions as follows:
+
+- 1:1 Renaming of attribute names or labels is allowed.
+- New attributes are allowed to be generated in a resource detector.
+
+## Internal details
+
+- Resource semantic conventions will need to be defined at the
+  `Resource Detector` level to ensure stability of identity of resources.
+- Attribute names and existing values cannot be renamed within stable versions
+  (pending OTEP #161)
+
+### New Semantic Conventions
+
+Going forward, Resource semantic conventions will be outlined at a "Resource
+detection" level.  While users can merge together resource detectors, each
+detector will provide a stable set of Resource attributes and labels.
```

Comment:
How do resource detectors differ from the current namespacing regime for resource attributes? Is `k8s.*` a resource detector? Is `k8s.cluster.*` a resource detector?